de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/api/impl/HUDDOrderBL.java

```diff
@@ -43,4 +60,53 @@ public void unassignHUs(final I_DD_OrderLine ddOrderLine, final Collection<I_M_H
 		final IHUDDOrderDAO huDDOrderDAO = Services.get(IHUDDOrderDAO.class);
 		huDDOrderDAO.removeFromHUsScheduledToMoveList(ddOrderLine, hus);
 	}
+
+	@Override
+	public void createBlockDDOrderForReceiptLines(final Properties ctx, final List<I_M_InOutLine> receiptLines)
+	{
+		final IHUAssignmentDAO huAssignmentDAO = Services.get(IHUAssignmentDAO.class);
+
+		final List<I_M_HU> husToDDOrder = new ArrayList<>();
+
+		for (final I_M_InOutLine receiptLine : receiptLines)
+		{
+			List<I_M_HU> topLevelHUsForReceiptLine = huAssignmentDAO.retrieveTopLevelHUsForModel(receiptLine);
+
+			husToDDOrder.addAll(topLevelHUsForReceiptLine);
+		}
+
+		createBlockDDOrderForHUs(ctx, husToDDOrder);
+
+	}
+
+	@Override
+	public void createBlockDDOrderForHUs(final Properties ctx, final List<I_M_HU> hus)
+	{
+		final IWarehouseDAO warehouseDAO = Services.get(IWarehouseDAO.class);
+		final IMsgBL msgBL = Services.get(IMsgBL.class);
+
+		final I_M_Warehouse blockWarehouse = warehouseDAO.retrieveBlockWarehouseOrNull();
+
+		if (blockWarehouse == null)
+		{
+			throw new AdempiereException(msgBL.getMsg(ctx, ERR_M_Warehouse_NoBlockWarehouse));
+		}
+
+		// run this out of trx
+
```

Comment:
I made this run out of trx because there is a safety check in HUs2DDOrderProducer.process() about it being so. de.metas.handlingunits.ddorder.api.impl.HUs2DDOrderProducer.process()